phpBB/phpbb/storage/adapter/local.php

```diff
@@ -138,12 +145,14 @@ public function delete($path)
 	{
 		try
 		{
-			$this->filesystem->remove($this->root_path . $path);
+			$this->filesystem->remove($this->root_path . $this->get_filepath($path));
 		}
 		catch (filesystem_exception $e)
 		{
 			throw new exception('STORAGE_CANNOT_DELETE', $path, array(), $e);
 		}
+
+		$this->remove_empty_dirs($path);
```

Comment:
I wonder if it might make sense to have this done via cron. Thoughts? (as in: let cron clean up paths every x hours)

Reply:
No, i don't think so, that can't be optimal when you have lot of files/dirs. It tries do remove empty directories from down to top while they are empty. Example:

rm /ab/cd/ef/file.jpg
rm /ab/cd/ef
rm /ab/cd
rm /ab

At least it should, i haven't tried it.